File: arrays_and_hashing/0001_two_sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        index = {}
        for idx, num in enumerate(nums):
            complement = target - num

            if complement in index:
                return [index[complement], idx]

            index[num] = idx

        return []

File: arrays_and_hashing/0001_two_sum.cpp
// https://leetcode.com/problems/two-sum/

#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> umap;

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
            if (umap.find(complement) != umap.end()) {
                return {umap[complement], i};
            }
            umap[nums[i]] = i;
        }
        return {};
    }
};
